### PR TITLE
Remove rcppparallel

### PR DESCRIPTION
**Pull request by koheiw**

### CODE REVIEW

**Comment by kbenoit** [comment]
So this removes the dependency on RcppParallel, and requires the user to install the (latest) TBB manually?


For Linux:
```
sudo apt install libtbb-dev
```

For macOS, assuming [homebrew](https://brew.sh/) is already installed, then [appears](https://formulae.brew.sh/formula/tbb) to be:
```
brew install tbb
```
(This worked on my Mac.)

And if TBB is not found, then it’s just non-parallel?

**Comment by koheiw** [comment]
Yes, it removes RcppParallel and uses more recent TBB library. 

Linux: the library needs to be installed beforehand (like the **curl** or **xml2** packages, that depend on header libraries). 
Windows: it uses static library from rtools (thanks to Thomas). 
Mac: not sure but hope "brew install tbb" works. Can you update Github Actions and SystemRequirements in DESCRIPTION for Mac?

**Comment by koheiw** [comment]
I am trying to bundle TBB but not successful...

https://github.com/quanteda/quanteda/tree/bundle-tbb

**Comment by kbenoit** [comment]
I can try this but currently I'm on holidays (until Monday).
Is there not a conditional so that if it is not detected as installed, it skips it?

**Comment by kbenoit** [comment]
The current version just failed on my mac:

```
   clang++ -arch arm64 -std=gnu++17 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -I../inst/include -DARMA_DONT_PRINT_OPENMP_WARNING -I'/Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/library/Rcpp/include' -I'/Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/library/RcppArmadillo/include' -I/opt/R/arm64/include    -fPIC  -falign-functions=64 -Wall -g -O2  -UNDEBUG -Wall -pedantic -g -O0 -fdiagnostics-color=always -c fcm.cpp -o fcm.o
   In file included from fcm.cpp:1:
   ../inst/include/lib.h:5:10: fatal error: 'oneapi/tbb.h' file not found
   #include <oneapi/tbb.h>
            ^~~~~~~~~~~~~~
   1 error generated.
   make: *** [fcm.o] Error 1
   ERROR: compilation failed for package ‘quanteda’
─
```

**Comment by koheiw** [comment]
`#include <oneapi/tbb.h>` might be just `#include <tbb/tbb.h>`. Please check how `brew` installs the library on Mac. I need to take a rest too!

**Comment by koheiw** [comment]
If `oneapi/tbb.h` is not found, serial code is used. These two are useful sources:

- https://cran.r-project.org/web/packages/Rcpp/vignettes/Rcpp-libraries.pdf
- https://github.com/corels/rcppcorels/tree/master

**Comment by kbenoit** [comment]
Thanks so much for your persistence on this tricky and deeply frustrating issue @koheiw.  

On macOS I'm seeing

```
❯ checking top-level files ... WARNING
  A complete check needs the 'checkbashisms' script.
  See section ‘Configure and cleanup’ in the ‘Writing R Extensions’
  manual.

❯ checking compilation flags in Makevars ... NOTE
  Package has both ‘src/Makevars.in’ and ‘src/Makevars’.
  Installation with --no-configure' is unlikely to work.  If you intended
  ‘src/Makevars’ to be used on Windows, rename it to ‘src/Makevars.win’
  otherwise remove it.  If ‘configure’ created ‘src/Makevars’, you need a
  ‘cleanup’ script.

0 errors ✔ | 1 warning ✖ | 1 note ✖
Error: R CMD check found WARNINGs
Execution halted

Exited with status 1.
```

File: src/Makevars.in [hunk 1]
```diff
@@ -0,0 +1,4 @@
+PKG_LIBS = $(LAPACK_LIBS) $(BLAS_LIBS) $(FLIBS) @TBB_LIBS@
```
Comment (kbenoit) [inline src/Makevars.in:1]: Should this file be deleted?

**Comment by koheiw** [comment]
‘src/Makevars.in’ is the template for ‘src/Makevars’, which is created by the configure script when the package is built: if oneapi/tbb.h is not found, the TBB flag is not defined. I have added the clean script already but not sure if it is working.

**Comment by koheiw** [comment]
You might need to `brew install checkbashisms` on your machine to suppress the warning.

Could you find a way to compile with TBB on Mac? You can check with `quanteda:::cpp_tbb_enabled()`.

**Comment by kbenoit** [comment]
So just so I'm clear, we now include TBB support, but instead of getting this code from **RcppParallel**, we require the user to install it themselves. So we should include instructions about this for each platform?

**Comment by koheiw** [comment]
Windows and (Mac?) users download binary files from CRAN, so they would not be affected. Linux or Unix users need to have the TBB header installed to enable parallel computing as in the SystemRequirements.

https://github.com/quanteda/quanteda/blob/cc214bcdfd4b56d74ef0b271f79ed1bacf2074c5/DESCRIPTION#L41

**Review by kbenoit — APPROVED** [review]
I submitted this branch to CRAN on 20 Mar. Maybe we should have merged it first and merged #2353 also??

**Pull request merged**